packages/react-storage/src/components/StorageBrowser/Views/Controls/AddFiles.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import { withBaseElementProps } from '@aws-amplify/ui-react-core/elements';
+
+import type { OmitElements } from '../types';
+import { StorageBrowserElements } from '../../context/elements';
+import { CLASS_BASE } from '../constants';
+import { useControl } from '../../context/controls';
+
+const { Button: ButtonElement } = StorageBrowserElements;
+
+const BLOCK_NAME = `${CLASS_BASE}__add-files`;
+
+export interface _AddFilesControl<
+  T extends StorageBrowserElements = StorageBrowserElements,
+> {
+  (): React.JSX.Element;
+  Button: T['Button'];
+}
+
+export interface AddFilesControl<
+  T extends StorageBrowserElements = StorageBrowserElements,
+> extends OmitElements<_AddFilesControl<T>, 'Button'> {
+  (): React.JSX.Element;
+}
+
+const Button = withBaseElementProps(ButtonElement, {
+  className: `${BLOCK_NAME}`,
+  variant: 'primary',
+});
+
+export const AddFilesControl: AddFilesControl = () => {
+  const [{ history }] = useControl({ type: 'NAVIGATE' });
+  const [{ selected }, handleUpdateState] = useControl({
+    type: 'ACTION_SELECT',
+  });
+
+  const { items: prevSelectedItems } = selected;
```

Comment:
Can improve readability by moving this destructuring closer to where `prevSelectedItems` is used